src/storage/mod.rs

```diff
@@ -251,19 +257,21 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                 // here.
                 let bypass_locks = TsSet::vec_from_u64s(ctx.take_resolved_locks());
 
-                // Update max_read_ts and check the in-memory lock table before getting the snapshot
-                concurrency_manager.update_max_read_ts(start_ts);
-                if ctx.get_isolation_level() == IsolationLevel::Si {
-                    concurrency_manager
-                        .read_key_check(&key, |lock| {
-                            Lock::check_ts_conflict(
-                                Cow::Borrowed(lock),
-                                &key,
-                                start_ts,
-                                &bypass_locks,
-                            )
-                        })
-                        .map_err(mvcc::Error::from)?;
+                if enable_async_commit {
+                    // Update max_read_ts and check the in-memory lock table before getting the snapshot
+                    concurrency_manager.update_max_read_ts(start_ts);
+                    if ctx.get_isolation_level() == IsolationLevel::Si {
+                        concurrency_manager
+                            .read_key_check(&key, |lock| {
+                                Lock::check_ts_conflict(
+                                    Cow::Borrowed(lock),
+                                    &key,
+                                    start_ts,
+                                    &bypass_locks,
+                                )
+                            })
+                            .map_err(mvcc::Error::from)?;
+                    }
```

Comment:
I extract those key checks to `async_commit_check_keys`.
Because we cannot borrow `Storage` to the spawned future, we need to pass quite a lot of arguments in :(